### PR TITLE
Runtime fixes.

### PR DESCRIPTION
**Pull request by Rohesie**
* Fixes runtimes related to xeno actions. In short it was removing the owner first and only after trying to unregister signals on a now `null` owner.

```
[18:39:41] Runtime in component.dm, line 135: Cannot read null.comp_lookup
proc name: UnregisterSignal (/datum/proc/UnregisterSignal)
src: Lunge (10) (/datum/action/xeno_action/activable/lunge)
call stack:
Lunge (10) (/datum/action/xeno_action/activable/lunge): UnregisterSignal(null, "warrior_used_grab")
Lunge (10) (/datum/action/xeno_action/activable/lunge): remove action(Young Warrior (904) (/mob/living/carbon/xenomorph/warrior))
Lunge (10) (/datum/action/xeno_action/activable/lunge): Destroy(0)
qdel(Lunge (10) (/datum/action/xeno_action/activable/lunge), 0)
Young Warrior (904) (/mob/living/carbon/xenomorph/warrior): set datum()
Young Warrior (904) (/mob/living/carbon/xenomorph/warrior): upgrade xeno("one")
Young Warrior (904) (/mob/living/carbon/xenomorph/warrior): update progression()
Young Warrior (904) (/mob/living/carbon/xenomorph/warrior): Life(2, 307)
Mobs (/datum/controller/subsystem/mobs): fire(0)
Mobs (/datum/controller/subsystem/mobs): ignite(0)
Master (/datum/controller/master): RunQueue()
Master (/datum/controller/master): Loop()
Master (/datum/controller/master): StartProcessing(0)
```
*Fixes a leader tracking runtime:
```
[22:02:24] Runtime in headset.dm, line 247: Cannot read null.squad_leader
proc name: disable sl direction (/obj/item/radio/headset/mainship/proc/disable_sl_direction)
usr: Smashlie/(Amala Khurana)
usr.loc: (Medical Lower (97, 70, 3))
src: the medical radio headset (/obj/item/radio/headset/mainship/doc)
src.loc: Amala Khurana (/mob/living/carbon/human)
call stack:
the medical radio headset (/obj/item/radio/headset/mainship/doc): disable sl direction()
the medical radio headset (/obj/item/radio/headset/mainship/doc): Topic("src=\[0x205358a];sl_direction=...", /list (/list))
Smashlie (/client): Topic("src=\[0x205358a];sl_direction=...", /list (/list), the medical radio headset (/obj/item/radio/headset/mainship/doc))
Smashlie (/client): Topic("src=\[0x205358a];sl_direction=...", /list (/list), the medical radio headset (/obj/item/radio/headset/mainship/doc))
```
* Fixes returning an atom on a signal, by using `.` as an auxiliary var.
* Fixes the medbot using old reagent IDs instead of paths.
* Fixes embedding not respecting the `EMBEDDEED_CAN_BE_YANKED_OUT` flag properly.
* Fixes a hallucination runtime:
```
[21:16:47] Runtime in unsorted.dm, line 37: to_chat called with invalid input type
proc name: stack trace (/datum/proc/stack_trace)
src: Chat (/datum/controller/subsystem/chat)
call stack:
Chat (/datum/controller/subsystem/chat): stack trace("to_chat called with invalid in...")
Chat (/datum/controller/subsystem/chat): queue(Horus \'Lupercal\' Briner (/mob/living/carbon/human), , 1)
to chat(Horus \'Lupercal\' Briner (/mob/living/carbon/human), , 1)
Aleric \'Klepto\' Caldwell (/obj/effect/fake_attacker): updateimage()
fake attack(Horus \'Lupercal\' Briner (/mob/living/carbon/human))
Horus \'Lupercal\' Briner (/mob/living/carbon/human): handle status effects()
Horus \'Lupercal\' Briner (/mob/living/carbon/human): handle status effects()
Horus \'Lupercal\' Briner (/mob/living/carbon/human): Life(2, 1762)
Horus \'Lupercal\' Briner (/mob/living/carbon/human): Life(2, 1762)
Horus \'Lupercal\' Briner (/mob/living/carbon/human): Life(2, 1762)
Mobs (/datum/controller/subsystem/mobs): fire(0)
Mobs (/datum/controller/subsystem/mobs): ignite(0)
Master (/datum/controller/master): RunQueue()
Master (/datum/controller/master): Loop()
Master (/datum/controller/master): StartProcessing(0)
```
*Fixes a machinegun runtime:
```

[21:27:11] Runtime in unsorted.dm, line 111: Cannot read null.loc
proc name: Get Angle (/proc/Get_Angle)
usr: Faeth1qwfhmukmghcd/(Violet Elliott)
usr.loc: (Alamo Landing Zone (141, 31, 2))
src: null
call stack:
Get Angle(the M56D mounted smartgun (/obj/machinery/m56d_hmg), null)
the M56D mounted smartgun (/obj/machinery/m56d_hmg): fire shot(Violet Elliott (/mob/living/carbon/human))
the M56D mounted smartgun (/obj/machinery/m56d_hmg): process shot(Violet Elliott (/mob/living/carbon/human))
the M56D mounted smartgun (/obj/machinery/m56d_hmg): InterceptClickOn(Violet Elliott (/mob/living/carbon/human), "icon-x=14;icon-y=6;left=1;scre...", the wooden barricade (/obj/structure/barricade/wooden))
Violet Elliott (/mob/living/carbon/human): check click intercept("icon-x=14;icon-y=6;left=1;scre...", the wooden barricade (/obj/structure/barricade/wooden))
Violet Elliott (/mob/living/carbon/human): ClickOn(the wooden barricade (/obj/structure/barricade/wooden), "icon-x=14;icon-y=6;left=1;scre...")
the wooden barricade (/obj/structure/barricade/wooden): Click(the floor (137,34,2) (/turf/open/floor/tile/white), "mapwindow.map", "icon-x=14;icon-y=6;left=1;scre...")
Faeth1qwfhmukmghcd (/client): Click(the wooden barricade (/obj/structure/barricade/wooden), the floor (137,34,2) (/turf/open/floor/tile/white), "mapwindow.map", "icon-x=14;icon-y=6;left=1;scre...")
```
* Fixes some issues caused by #2913 and associated runtimes:
```
[04:20:48] Runtime in abilities.dm, line 808: Cannot execute null.visible message().
proc name: action activate (/datum/action/xeno_action/lay_egg/action_activate)
usr: CKEY/(Elder Shrike)
usr.loc: (Civilian Residences North (52, 150, 2))
src: Lay Egg (400) (/datum/action/xeno_action/lay_egg)
call stack:
Lay Egg (400) (/datum/action/xeno_action/lay_egg): action activate()
Lay Egg (/obj/screen/action_button): Click(null, "mapwindow.map", "icon-x=20;icon-y=11;left=1;scr...")
```

### CODE REVIEW

File: code/datums/actions/xeno_action.dm [hunk 1]
```diff
@@ -211,6 +217,9 @@
 
 //override this
 /datum/action/xeno_action/activable/proc/can_use_ability(atom/A, silent = FALSE, override_flags)
+	if(QDELETED(owner))
```
Comment (spookydonut) [inline code/datums/actions/xeno_action.dm]: if the owner has been qdeleted then you've got bigger issues, this is problem hiding.

File: code/game/objects/items/radio/headset.dm [hunk 2]
```diff
@@ -244,7 +244,7 @@ GLOBAL_LIST_INIT(channel_tokens, list(
 	if(wearer.mind && wearer.assigned_squad && wearer.hud_used?.SL_locator)
 		wearer.hud_used.SL_locator.alpha = 0
 
-	if(wearer.assigned_squad.squad_leader == wearer)
+	if(wearer.assigned_squad.squad_leader && wearer.assigned_squad.squad_leader == wearer)
```
Comment (spookydonut) [inline code/game/objects/items/radio/headset.dm]:
```suggestion
	if(wearer?.assigned_squad?.squad_leader == wearer)
```

File: code/datums/actions/action.dm [hunk 3]
```diff
@@ -81,6 +81,7 @@
 	M.update_action_buttons()
 	M.actions_by_path[type] = src
 
+
```
Comment (spookydonut) [inline code/datums/actions/action.dm]: ???

File: code/datums/actions/xeno_action.dm [hunk 4]
```diff
@@ -99,8 +101,11 @@
 
 	return TRUE
 
+
 /datum/action/xeno_action/fail_activate()
-	update_button_icon()
+	if(button)
```
Comment (spookydonut) [inline code/datums/actions/xeno_action.dm]: should always have a button

File: code/game/objects/effects/landmarks/landmarks.dm [hunk 5]
```diff
@@ -56,10 +56,8 @@
 		qdel(src)
 
 
-/obj/effect/landmark/newplayer_start/Initialize()
-	. = ..()
+/obj/effect/landmark/newplayer_start/New() //This can't be Initialize() or players will start in a wrong loc at roundstart.
```
Comment (spookydonut) [inline code/game/objects/effects/landmarks/landmarks.dm]: change it to do `GLOB.newplayer_start += src` then the newplayer code to do `get_turf(pick(GLOB.newplayer_start))`

File: code/datums/datum.dm [hunk 6]
```diff
@@ -173,7 +173,7 @@
 
 
 /datum/proc/can_interact(mob/user)
-	if(!user.can_interact_with(src))
+	if(QDELETED(user) || !user.can_interact_with(src)) //This is used by Topic and can thus be called with an outdated or inexistent user.
```
Comment (spookydonut) [inline code/datums/datum.dm]: if it's used by topic then topic would be using usr

File: code/game/objects/effects/landmarks/landmarks.dm (hunk 5 above)
Comment (Rohesie) [reply code/game/objects/effects/landmarks/landmarks.dm]: Done.

File: code/modules/mob/living/carbon/xenomorph/xenomorph.dm [hunk 7]
```diff
@@ -61,9 +61,6 @@
 	speed = xeno_caste.speed
 	armor = getArmor(arglist(xeno_caste.armor))
 
-	for(var/i in actions)
-		qdel(i)
-	actions = xeno_caste.actions.Copy()
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/xenomorph.dm]: removing this will break abilities

**Pull request merged**